### DISCUSSION TITLE
Issue with FN+F10 disabling touchpad on ASUS FA507XV and causing unintended screenshot and screen brightness behavior

### DISCUSSION THREAD

**Discussion by YiNMiaoN** [seerge/g-helper#1504] [Q&A] [answered]
I am encountering an issue with my ASUS FA507XV laptop, specifically when using the FN+F10 key combination to disable the touchpad. Whenever I use this shortcut, it not only disables the touchpad as intended but also takes a screenshot of the screen. Additionally, the screen brightness temporarily decreases and then returns to normal.

**Comment by seerge** [comment]
Hello, it looks more like a hardware issue with your keyboard emitting wrong keys? :)

**Reply by YiNMiaoN** [reply]
Thank you for your response, but I have confirmed that this issue only occurs when the G-Helper software is running. The touchpad can be disabled properly when the software is not running, and the aforementioned issue does not occur.

**Reply by seerge** [reply]
Can you post a log from pressing Fn+F10 ? Do you have Asus services running in background ?

App would just emit ``Ctrl + Win + F24`` when you press Fn+F10, which stands for windows default combination for touchpad toggling :) and show appropriate toast notification. There are no other actions next to that

**Reply by YiNMiaoN** [reply]
I also wanted to find the log files, but I didn't see any detailed logs regarding the use of the shortcut key in the logs. :）

**Reply by seerge** [reply]
Logs are in ``%appdata%/ghelper``

**Comment by MC-dusk** [comment]
Oh, I have exactly the same trouble, and it costs me a whole afternoon to find what's wrong with my PC...
I have never thought it has relation with the g-helper until this post...

**Reply by YiNMiaoN** [reply]
It's great to find someone with the same issue, as it indicates that this bug can be reproduced. I believe it might be because the author adapted the software for a specific Lenovo model. I have searched online and found screenshots showing that the Fn+F10 combination is assigned as a screenshot shortcut on Lenovo laptops. It's possible that on our computers, this shortcut has not been disabled.

**Reply by MC-dusk** [reply]
Let me say what I've found. The screen brightness's temporarily decreasing is because the system receives some keyboard inputs like "win+prtsc", which is the windows's default screenshot shortcut. You can press it manully and it has the same effect. I used OBS input overlay to prove that, the input history showed that when I pressed "fn+f10", what I actually input was "lctrl+rctrl+win+f24", and it somehow triggered the screenshot.
<img width="290" alt="image" src="https://github.com/seerge/g-helper/assets/52311860/33c3e436-f927-49dd-b47e-9d6a9dbee4e7">
You can check the "C:\Users\user\Pictures\Screenshots" folder, I guess there are a bunch of screenshots.

**Comment by seerge** [comment]
@YiNMiaoN @MC-dusk 

I have replaced ``Ctrl + Win + F24`` command with ``Win + LeftControl + F24`` (apparently ``LeftControl`` it exists as a separate "key")

Check if it makes difference for you

[GHelper.zip](https://github.com/seerge/g-helper/files/13062868/GHelper.zip)

**Reply by MC-dusk** [reply]
There is no Rctrl anymore, but the screenshot still exists.
The screenshot isn't triggered 100%.
This time windows start menu will pop up sometimes after pressing the key combination.

**Reply by MC-dusk** [reply]
If I use the fn-lock, then everything goes right.
<img width="262" alt="image" src="https://github.com/seerge/g-helper/assets/52311860/23d62151-7f9a-42ca-a5c4-fd74e12d6d7b">

**Reply by YiNMiaoN** [reply]
I tried the file you provided, but it doesn't work. The issue has not been resolved. :<

**Reply by seerge** [reply]
@MC-dusk do you have AsusOptimization service running in the background ? cause it will also produce some actions ...

I have changed the order of keys get unpressed (after being pressed), mb it will make a difference

[GHelper.zip](https://github.com/seerge/g-helper/files/13062919/GHelper.zip)

Just in case for me it always works :) on 2 separate asus laptops ..

**Reply by YiNMiaoN** [reply]
Thank you. It is now functioning as expected, with the touchpad being disabled instead of capturing screenshots.

**Reply by YiNMiaoN** [reply]
This issue reoccurs after I restart, and I tried reinstalling the software, but it doesn't work.

**Reply by MC-dusk** [reply]
reply to this version: https://github.com/seerge/g-helper/discussions/1504#discussioncomment-7350327

I think it does work in some way, though there still are some screenshots sometimes, and seems to be no obvious law. The screenshot frequency has significantly decreased, I feel.
The win start menu poping up disappeared. haha

**Reply by seerge** [reply]
@MC-dusk do you have asus services running?

**Reply by MC-dusk** [reply]
> @MC-dusk do you have asus services running?

No, I turned it off because it would cause conflicts with g-helper's hotkey and screen display.

**Reply by seerge** [reply]
@MC-dusk i see. Well, I can only imagine that  in your case windows just interprets ``Win + Ctrl + F24`` (touchpad toggle) with ``Win+F24`` (printscreen)

It's a stupid hotkey selection in windows itself in a first place (same keys) . I can't really offer anything here

https://www.reddit.com/r/Windows10/comments/nv5cdr/hidden_windows_key_hotkeys/

**Reply by MC-dusk** [reply]
There were always some screenshots before. The frequency was low, so I didn't realize it. After some weird Windows or GHelper updates recently, maybe, the frequency has risen, so I noticed it. This modified version does have some improvements: the frequency of screenshots has reduced, I feel, if not press the key quickly. So I think it's nice enough.

For me, if I can completely solve it, it's the best. If I can't, it's okay, after all, I won't frequently switch touchpads.

**Comment by Oleg52** [comment]
I have the same issue on my fx517, i tried to add 100ms delay in `KeyKeyKeyPress` method before firing key up events and it fixed the issue, touchpad toggles as it should and screenshot is not being taken. But is it fine to be fixed in that way? 😅 As long as it doesn't break custom key actions.

**Comment by seerge** [comment, marked as answer]
@YiNMiaoN @MC-dusk and others in this thread,

Apparently on FA507 laptop Fn+F10 emits the touchpad toggle on hardware level. So when g-helper was doing it next to hardware - it was emitting it twice

Someone had a similar problem https://github.com/seerge/g-helper/issues/1589 
You can try build from that thread and see if it works for you 

https://github.com/seerge/g-helper/files/13328822/GHelper.zip

**Reply by YiNMiaoN** [reply]
Thank you, it is working perfectly now without any unexpected issues. This problem has been resolved perfectly.

**Reply by MC-dusk** [reply]
Nice, and the problem is just as I guess. As the [YiNMiaoN](https://github.com/YiNMiaoN) said [above](https://github.com/seerge/g-helper/discussions/1504#discussioncomment-7350070):

> Thank you for your response, but I have confirmed that this issue only occurs when the G-Helper software is running. The touchpad can be disabled properly when the software is not running, and the aforementioned issue does not occur.

[And I've turned off the asus services.](https://github.com/seerge/g-helper/discussions/1504#discussioncomment-7354501)

> No, I turned it off because it would cause conflicts with g-helper's hotkey and screen display.

Without the g-helper and asus services, then it must be the hardware signal.
